### PR TITLE
Discovery config relations

### PR DESCRIPTION
**Pull request by benbosman**
This is a configuration of additional search filters used in the UI to display related items.
This is used in 
https://github.com/DSpace/dspace-angular/pull/340 and was discovered to be missing in the comment 
https://github.com/DSpace/dspace-angular/pull/340#issuecomment-453184749

### CODE REVIEW

**Review by tdonohue — APPROVED** [review]
👍 Looks like an obvious, minor configuration change.  It is just new configs with corresponding Integration Test initialization.

File: dspace/config/spring/api/discovery.xml [hunk 1]
```diff
@@ -254,6 +259,11 @@
                 <ref bean="searchFilterFileNameInOriginalBundle" />
                 <ref bean="searchFilterFileDescriptionInOriginalBundle" />
                 <ref bean="searchFilterEntityType"/>
+                <ref bean="searchFilterIsAuthorOfPublicationRelation"/>
+                <ref bean="searchFilterIsProjectOfPublicationRelation"/>
+                <ref bean="searchFilterIsOrgUnitOfPublicationRelation"/>
+                <ref bean="searchFilterIsPublicationOfJournalIssueRelation"/>
+		        <ref bean="searchFilterIsJournalOfPublicationRelation"/>
```
Comment (AlexanderS) [inline dspace/config/spring/api/discovery.xml]: Just a very minor nitpick: Can we fix the indentation here?

**Comment by benbosman** [comment]
I've merged the latest configurable_entities codebase in. I've also disabled the entities ITs because Travis is failing on them (the full test does succeed locally)

So in summary, the differences compare to the state reviewed by @AlexanderS yesterday are:
* The indentation has been changed
* The latest configurable_entities has been merged
* The entities ITs are disabled for Travis (I will always enable them locally to ensure no erroneous code gets committed)

File: dspace/config/spring/api/discovery.xml [hunk 2]
```diff
@@ -119,6 +119,11 @@
                 <ref bean="searchFilterFileNameInOriginalBundle" />
                 <ref bean="searchFilterFileDescriptionInOriginalBundle" />
                 <ref bean="searchFilterEntityType"/>
+                <ref bean="searchFilterIsAuthorOfPublicationRelation"/>
+                <ref bean="searchFilterIsProjectOfPublicationRelation"/>
+                <ref bean="searchFilterIsOrgUnitOfPublicationRelation"/>
+                <ref bean="searchFilterIsPublicationOfJournalIssueRelation"/>
+	            <ref bean="searchFilterIsJournalOfPublicationRelation"/>
```
Comment (paulo-graca) [inline dspace/config/spring/api/discovery.xml]: identation fix required.

File: dspace/config/spring/api/discovery.xml [hunk 3]
```diff
@@ -254,6 +259,11 @@
                 <ref bean="searchFilterFileNameInOriginalBundle" />
                 <ref bean="searchFilterFileDescriptionInOriginalBundle" />
                 <ref bean="searchFilterEntityType"/>
+                <ref bean="searchFilterIsAuthorOfPublicationRelation"/>
+                <ref bean="searchFilterIsProjectOfPublicationRelation"/>
+                <ref bean="searchFilterIsOrgUnitOfPublicationRelation"/>
+                <ref bean="searchFilterIsPublicationOfJournalIssueRelation"/>
+		<ref bean="searchFilterIsJournalOfPublicationRelation"/>
```
Comment (paulo-graca) [inline dspace/config/spring/api/discovery.xml]: identation fix required.

**Review by AlexanderS — APPROVED** [review]
Looks perfekt now.

**Comment by benbosman** [comment]
2 approvals => merging

**Pull request merged**